evm_arithmetization/src/cpu/kernel/asm/mpt/linked_list/initial_tries.asm

```diff
@@ -0,0 +1,187 @@
+    JUMP
+
+set_payload_extension:
+    // stack: node_type, after_node_type, storage_ptr_ptr, retdest
+    POP
+    // stack: after_node_type, storage_ptr_ptr, retdest
+    %add_const(2) %mload_trie_data
+    // stack: child_ptr, after_node_type, storage_ptr_ptr, retdest
+    %jump(mpt_set_payload)
+set_payload_storage_extension:
+    // stack: node_type, after_node_type, account_ptr_ptr, storage_ptr_ptr, retdest
+    POP
+    // stack: after_node_type, account_ptr_ptr, storage_ptr_ptr, retdest
+    %add_const(2) %mload_trie_data
+    // stack: child_ptr, account_ptr_ptr, storage_ptr_ptr, retdest
+    %jump(mpt_set_storage_payload)
+
+set_payload_leaf:
+    // stack: node_type, after_node_type, account_ptr_ptr, storage_ptr_ptr, retdest
+    POP
+    %add_const(2) // The payload pointer starts at index 3, after num_nibbles and packed_nibbles.
+    DUP1 
+    // stack payload_ptr_ptr, payload_ptr_ptr, account_ptr_ptr, storage_ptr_ptr, retdest
+    %mload_trie_data 
+    // stack account_ptr, payload_ptr_ptr, account_ptr_ptr, storage_ptr_ptr, retdest
+    %add_const(2)
+    %mload_trie_data // storage_root_ptr = account[2]
+
+    DUP1 %mload_trie_data
+    POP
+    // stack storage_root_ptr, payload_ptr_ptr, account_ptr_ptr, storage_ptr_ptr, retdest
+    %stack
+        (storage_root_ptr, payload_ptr_ptr, account_ptr_ptr, storage_ptr_ptr) ->
+        (storage_root_ptr, storage_ptr_ptr, after_set_storage_payload, storage_root_ptr, payload_ptr_ptr, account_ptr_ptr)
+    %jump(mpt_set_storage_payload)
+after_set_storage_payload:
+    // stack: storage_ptr_ptr', storage_root_ptr, payload_ptr_ptr, account_ptr_ptr, retdest
+    DUP4
+    MLOAD_GENERAL // load the next payload pointer in the linked list
+    DUP1 %add_const(2) // new_storage_root_ptr_ptr = payload_ptr[2]
+    // stack: new_storage_root_ptr_ptr, new_payload_ptr, storage_root_ptr, storage_ptr_ptr', payload_ptr_ptr, account_ptr_ptr, retdest
+    // Load also the old "dynamic" payload for storing the storage_root_ptr
+    DUP6 %decrement
+    MLOAD_GENERAL
+    %add_const(2) // dyn_storage_root_ptr_ptr = dyn_paylod_ptr[2]
+    %stack
+        (dyn_storage_root_ptr_ptr, new_storage_root_ptr_ptr, new_payload_ptr, storage_ptr_ptr_p, storage_root_ptr, payload_ptr_ptr, account_ptr_ptr) ->
+        (new_storage_root_ptr_ptr, storage_root_ptr, dyn_storage_root_ptr_ptr, storage_root_ptr, payload_ptr_ptr, new_payload_ptr, account_ptr_ptr, storage_ptr_ptr_p)
+    %mstore_trie_data // The initial account pointer in the linked list has no storage root so we need to manually set it.
+    %mstore_trie_data // The dynamic account pointer in the linked list has no storage root so we need to manually set it.
+    %mstore_trie_data // Set the leaf payload pointing to next account in the linked list.
+    // stack: account_ptr_ptr, storage_ptr_ptr', retdest
+    %add_const(@STORAGE_NEXT_NODE_PTR) // The next pointer is at distance `STORAGE_NEXT_NODE_PTR`
```

Comment:
The name `@STORAGE_NEXT_NODE_PTR` confuses me. We want to advance by `@ACCOUNTS_LINKED_LISTS_NODE_SIZE`, don't we? I don't see the link with storage.